physics physx docs (#1633)
cocos-creator/creator-docs: en/physics/physics-item.md

## Physics Engine: bullet(ammo.js)

[ammo.js](https://github.com/cocos-creator/ammo.js) is the __asm.js__/__wasm__ Version of the [bullet](https://github.com/bulletphysics/bullet3) physics engine, it is compiled by [emscripten](https://github.com/emscripten-core/emscripten) tool. Bullet has perfect physical functions, and we will put more work here in the future.

It should be noted that currently the __ammo.js__ module has a size of about __1.5MB__.

## Physics Engine: PhysX

**PhysX**（[GitHub](https://github.com/NVIDIAGameWorks/PhysX)）is an open source real-time commercial physics engine developed by Nvidia. It has perfect functional characteristics and extremely high stability, as well as excellent performance. 

The **PhysX** currently supported in Cocos Creator is version 4.1 and is allowed to be used on most platforms. However, due to the large size of **PhysX** and some limitations of its own, there are some platforms Unable to get good support, these platforms are:

- Various Mini Games platforms with package body restrictions (the reason is that the package body of the **PhysX** library is about 5 MB)
- Android x86 device

> Some newer platforms, such as HarmonyOS and Apple M1 devices, will be supported in the future. Please pay attention to the update announcement. In addition, the ByteDance platform provides the underlying native physics, so this module can also be run in ByteDance Mini Games. For specific details, please refer to [publish ByteDance Mini Games](../editor/pulish/../publish/publish-bytedance-mini-game.md).

Reviewer: I believe this link is wrong?

[publish ByteDance Mini Games](../editor/pulish/../publish/publish-bytedance-mini-game.md).

This looks incorrect.

> When publishing to the native platform, it is strongly recommended to use this module, so that you can get good performance, especially the native application of the IOS system.

Reviewer: ```suggestion
When publishing to the native platform, it is strongly recommended to use this module, to get good performance, especially the native application of the IOS system.
```

> ## Do Not Use Physics

Reviewer: ```suggestion
## Turning Off Physics
```

> If you don't need to use any physics related components and interfaces, you can uncheck the yellow box, so that you can get a smaller package when publishing.

Reviewer: ```suggestion
To turn off physics and related components and interfaces, uncheck the yellow box, which helps lessen the package size when publishing.
```

> **Note**: if it is unchecked, the project will not be able to use physics related components and interfaces, otherwise an error will be reported during runtime.

<!-- ## Expand the physical backend -->

# Physics Engine Performance

Mainly for various Mini Games platforms and native platforms, and compare **Bullet** and **PhysX**:

- On native and ByteDance platforms, using **PhysX** can get the best performance
- On various Mini Games platforms, use **Bullet** to get the best performance

# Physics Engine Differences

Different physics engines have different internal designs and algorithms, so there will be some different behaviors under the same parameters. These differences are mainly:

- `damping` property on RigidBody components
- `factor` property on RigidBody components
- PhysicsMaterial

Difference details and some solutions:

1. `damping`
The damping term is different because PhysX uses different damping algorithm models.

> However, this difference has been eliminated internally. If you need to synchronize the damping value previously set in PhysX, you can refer to the following code to convert:

Reviewer: ```suggestion
However, this difference has been eliminated internally. To synchronize the damping value previously set in PhysX, refer to the following code to convert:
```
